### DISCUSSION TITLE
TTGO T-Embed Project

### DISCUSSION THREAD

**Discussion by wizmo2** [sle118/squeezelite-esp32#311] [Show and tell]
I'm currently working a new project with the [TTGO T-Embed](https://github.com/wizmo2/TAudio-Case/tree/main/tembed)

![image](https://github.com/sle118/squeezelite-esp32/assets/11602094/fe59af74-f393-4973-b799-b8fdd035b1bf)

Its work in progress, but the kit looks like an ideal candidate for the next generation Squeezelite-ESP32 players.  It comes with 
- ESP32-S3 
- MAX98357A mono DAC with amp
- ST7789 320x170 SPI display
- Rotary dial with button
- LED ring light

My 3D printed case design accommodates a SPDIF output and IR sensor.

I have managed to get it fully functional for dac and spdif output.  I have a branch in my fork which includes various mods to support the non-standard ST7789 resolution and the APA102 led ring.  There's also some other patches for ESP32S3 variations, but it all needs cleaning up.

@philippe44, how do you want to handle ongoing S3 development?  For example, do you want me to prepare PRs to the main branch, the v4.4-s3 branch, or just leave you to it!

**Comment by philippe44** [comment]
Looks very nice indeed! With regard to branches, currently I'm trying to maintain a unique source code set for esp32 4.x and esp32s3 4.4. What is committed today works on 4.3 and 4.4 and esp32 and esp32-s3

**Comment by wizmo2** [comment]
I'm more-or-less done with the T-Embed S3 migration.  If anyone wants to give it a go there is a downloader for my current build here (https://wizmo2.github.io/TAudio-Case/)

I just have USB-JTAG console inputs to figure out (@philippe44, did you make any progress on this.  I have the stout working, but need to look into stdin some more)

Next steps are:
- PR some of the modifications, so my fork doesn't drift too far from the master.  
- I have a mini W5500 ethernet module on order, that should fit in the existing case.
